modules/transactions/src/main/java/org/apache/ignite/internal/tx/message/TxFinishReplicaRequest.java

```diff
@@ -59,6 +59,13 @@ public interface TxFinishReplicaRequest extends PrimaryReplicaRequest, Timestamp
      */
     long commitTimestampLong();
 
+    /**
+     * Get the transaction coordinator inconsistent ID.
+     *
+     * @return the transaction coordinator inconsistent ID
+     */
+    String txCoordinatorId();
```

Comment:
I mean logically. Why do we need coordinatorId on finish request? From my point of view, we use coordinatorId retrieved from txnState local map in two cases.

1. In order to check whether coordinator is dead and trigger the recovey, which definitely has no sense in case of finished transaction.
2. In order to decrement counter of an "in-flight" transactions that is used within index build procedure. In that case it's guaranteed that txnState local map will be populated with txCoordId prior to finish processing.

All in all, I believe that we may remove txCoordinatorId() from TxFinishReplicaRequest and update 
```
    private void markFinished(UUID txId, boolean commit, @Nullable HybridTimestamp commitTimestamp) {
        txManager.updateTxMeta(txId, old -> new TxStateMeta(
                commit ? COMMITTED : ABORTED,
                old == null ? null : old.txCoordinatorId(),
...
```